lib/charms/archive_auth_mirror/repository.py

```diff
@@ -44,3 +48,21 @@ def split_repository_uri(uri):
     """Split the repository URI into components."""
     parts = ('url', 'suite', 'components')
     return dict(zip(parts, uri.split(' ', maxsplit=2)))
+
+
+def update_config(get_paths=get_paths, suite=None, sign_key_id=None,
```

Comment:
we should probably move this to utils.py since it's not just for reprepro now, but it can be done in another branch